github/CVSS.py

```diff
@@ -0,0 +1,57 @@
+############################ Copyrights and license ############################
+#                                                                              #
+# Copyright 2023 crimsonknave <crimsonknave@github.com>                        #
+#                                                                              #
+# This file is part of PyGithub.                                               #
+# http://pygithub.readthedocs.io/                                              #
+#                                                                              #
+# PyGithub is free software: you can redistribute it and/or modify it under    #
+# the terms of the GNU Lesser General Public License as published by the Free  #
+# Software Foundation, either version 3 of the License, or (at your option)    #
+# any later version.                                                           #
+#                                                                              #
+# PyGithub is distributed in the hope that it will be useful, but WITHOUT ANY  #
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS    #
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more #
+# details.                                                                     #
+#                                                                              #
+# You should have received a copy of the GNU Lesser General Public License     #
+# along with PyGithub. If not, see <http://www.gnu.org/licenses/>.             #
+#                                                                              #
+################################################################################
+
+from decimal import Decimal
+from typing import Any, Dict
+
+from github.GithubObject import Attribute, NonCompletableGithubObject, NotSet
+
+
+class CVSS(NonCompletableGithubObject):
+    """
+    This class represents a CVSS.
+    The reference can be found here <https://docs.github.com/en/rest/security-advisories/global-advisories>
+    """
+
+    def _initAttributes(self) -> None:
+        self._vector_string: Attribute[str] = NotSet
+        self._score: Attribute[Decimal] = NotSet
+
+    @property
+    def score(self) -> Decimal:
```

Comment:
Why not float?

Reply:
Fundamentally the score is a user facing number so having it be a decimal means that it will always look nice. There won't be a CVSS score that needs more than 1-2 decimal places, so we don't need any of the float precision.

Additionally, decimal will ensure that we don't end up with 5.00000000000001 (or whatever) when doing any math or converting with a score. It also means I can be confident that any comparisons I do against the score won't be impacted by the same issue (so, if I check if the score is 5 less for example, the above float value.

The [decimal docs](https://docs.python.org/3/library/decimal.html) go into more detail into the pros and cons.

Reply:
Sounds good, thanks!